packages/dev/core/src/Lights/Shadows/shadowGenerator.ts

```diff
@@ -864,14 +879,17 @@ export class ShadowGenerator implements IShadowGenerator {
      * @param usefullFloatFirst By default the generator will try to use half float textures but if you need precision (for self shadowing for instance), you can use this option to enforce full float texture.
      * @param camera Camera associated with this shadow generator (default: null). If null, takes the scene active camera at the time we need to access it
      * @param useRedTextureType Forces the generator to use a Red instead of a RGBA type for the shadow map texture format (default: false)
+     * @param forceGLSL defines a boolean indicating if the shader must be compiled in GLSL even if we are using WebGPU
      */
-    constructor(mapSize: number, light: IShadowLight, usefullFloatFirst?: boolean, camera?: Nullable<Camera>, useRedTextureType?: boolean) {
+    constructor(mapSize: number, light: IShadowLight, usefullFloatFirst?: boolean, camera?: Nullable<Camera>, useRedTextureType?: boolean, forceGLSL = false) {
         this._mapSize = mapSize;
         this._light = light;
         this._scene = light.getScene();
         this._camera = camera ?? null;
         this._useRedTextureType = !!useRedTextureType;
 
+        this._initShaderSourceAsync(forceGLSL);

```

Comment:
Shouldn't we make this call only if `this._shadersLoaded == false`, to avoid running the `await Promise.all` each time?